app/kubernetes/store/namespace.js

```diff
@@ -0,0 +1,20 @@
+import _ from 'lodash-es';
```

Comment:
IMO it could have been easily integrated into the namespaceService. I think it's not a bad idea to couple the fetching and storing of the data.

if we are going in the way of just a regular es6 file, a class is not needed (since you're exposing an instance).

```js
let namespaces = [];

export function getNamespaces() {
  return [...namespaces] (to prevent mutating the data)
}

export function updateNamespaces(value = []) {
  namespaces = value
}

```

Reply:
the problem with going to non angular way is we remove things like value binding (when namespaces change, how do angular knows to rerender?)

Reply:
I was first thinking of integrating it to the `NamespaceService` but as it would be inside an AngularJS instance, it would require to integrate `NamespaceHelper` in AngularJS (`angular.service()` or `angular.factory()`), making it accessible only through DI, which is against its design as it should not be an instance (Helper containing only static functions).

`updateNamespaces` should only be called from `NamespaceService.get()`, which triggers a $digest cycle. All other values will update with the latest values on rerender

Reply:
and what about exporting just  few functions vs a class instance?